presto-orc/src/main/java/com/facebook/presto/orc/reader/DoubleSelectiveStreamReader.java

```diff
@@ -247,6 +292,60 @@ private int readNoFilter(int[] positions, int positionCount)
     private int readWithFilter(int[] positions, int positionCount)
             throws IOException
     {
+        boolean testNull = (nonDeterministicFilter && this.filter.testNull()) || nullsAllowed;
+
+        if (useBatchMode()) {
+            int totalPositionCount = positions[positionCount - 1] + 1;
+            int readCount = 0;
+
+            final int filteredPositionCount;
+
+            if (presentStream == null) {
```

Comment:
`useBatchMode()` presents the `presentStream == null` condition. Does it still make sense to double-check the same condition here?

Reply:
@sdruzkin Thank you for the question.The float and double batch mode was set only when presentStream == null because of the extra int <-> float long <-> double conversion. I think it would be less error prone to keep it the same structure as other types.